src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -111,34 +106,70 @@ public static TensorFlowTransform Create(IHostEnvironment env, ModelLoadContext
+            bool isFrozen = true;
+            bool isNonFrozenModelSupported = ctx.Header.ModelVerReadable >= 0x00010003;
+            if (isNonFrozenModelSupported)
+                isFrozen = ctx.Reader.ReadBoolByte();
 
-            bool isMultiOutput = ctx.Header.ModelVerReadable >= 0x00010002;
-            var numOutputs = 1;
-            if (isMultiOutput)
-                numOutputs = ctx.Reader.ReadInt32();
+            if (isFrozen)
+            {
+                byte[] modelBytes = null;
+                if (!ctx.TryLoadBinaryStream("TFModel", r => modelBytes = r.ReadByteArray()))
+                    throw env.ExceptDecode();
 
-            env.CheckDecode(numOutputs > 0);
-            var outputs = new string[numOutputs];
-            for (int j = 0; j < outputs.Length; j++)
-                outputs[j] = ctx.LoadNonEmptyString();
+                var io = ModelInputsOutputs(env, ctx);
+                return new TensorFlowTransform(env, modelBytes, io.Item1, io.Item2, isFrozen);
+            }
+            else
+            {
+                Dictionary<string, byte[]> savedModel = new Dictionary<string, byte[]>();
+                var load = ctx.TryLoadBinaryStream("TFSavedModel", br =>
+                {
+                    int count = br.ReadInt32();
+                    for (int n = 0; n < count; n++)
+                    {
+                        string key = br.ReadString();
+                        byte[] value = br.ReadByteArray();
+                        savedModel.Add(key, value);
+                    }
+                });
+
+                var tempDirPath = Path.GetFullPath(Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()));
+                var tempDirInfo = Directory.CreateDirectory(tempDirPath);
+                foreach (var kvp in savedModel)
+                {
+                    string fileName = Path.Combine(tempDirInfo.FullName, kvp.Key);
+                    if (kvp.Key.StartsWith(SavedModelVariablesDirName))
+                    {
+                        var variabledDirInfo = new DirectoryInfo(Path.Combine(tempDirPath, SavedModelVariablesDirName));
+                        if (!variabledDirInfo.Exists)
+                            Directory.CreateDirectory(variabledDirInfo.FullName);
 
-            return new TensorFlowTransform(env, modelBytes, inputs, outputs);
+                        var tokens = kvp.Key.Split(Path.DirectorySeparatorChar);
+                        fileName = Path.Combine(variabledDirInfo.FullName, tokens[1]);
+                    }
+
+                    using (var fs = new FileStream(fileName, FileMode.Create, FileAccess.Write))
+                    {
+                        fs.Write(kvp.Value, 0, kvp.Value.Length);
+                    }
+                }
+
+                var session = GetSession(env, tempDirPath);
+                var io = ModelInputsOutputs(env, ctx);
+                Directory.Delete(tempDirPath, true);
```

Comment:
>Directory.Delete(tempDirPath, true); [](start = 16, length = 36)

Nice to do it in finally block so that in case exception is thrown in between creation and deletion, directory is always deleted.